crates/prompt_gateway/src/hallucination.rs

```diff
@@ -18,9 +18,11 @@ pub fn extract_messages_for_hallucination(messages: &Vec<Message>) -> Vec<String
         for message in messages.iter().rev() {
             if let Some(model) = message.model.as_ref() {
                 if !model.starts_with(ARCH_MODEL_PREFIX) {
-                    if message.role == ASSISTANT_ROLE {
-                        break;
-                    }
+                  if let Some(content) = &message.content {
+                      if !content.starts_with(HALLUCINATION_TEMPLATE) {
+                          break;
+                      }
+                  }
```

Comment:
is this needed

Reply:
Yes, this is for separating the hallucination template. When we collect the user history message, we want to go pass over the chat template if there is parameter gathering from ArchFC.

Reply:
Yes, this is for separating the hallucination template. When we collect the user history message, we want to go pass over the chat template if there is parameter gathering from ArchFC.

Reply:
can you add test case to simulate this behavior

Reply:
the current test case reflects this.